src/server/src/main/java/io/cassandrareaper/storage/IStorage.java

```diff
@@ -90,6 +90,8 @@
 
   boolean updateRepairSegment(RepairSegment newRepairSegment);
 
+  boolean updateRepairSegmentUnsafe(RepairSegment newRepairSegment);
```

Comment:
The CassandraStorage class was the only one requiring that we had the lead before updating the repair segment.
I think that this should apply to PostgresStorage as well though and then agree this should be part of `IDistributedStorage` instead of `IStorage`.

Reply:
I made it so that `PostgresStorage` also checks that the current instance can renew the lead before updating the segment.